ui/src/components/SampleView/MotorControls.js

```diff
@@ -16,6 +16,10 @@ export default class MotorControls extends React.Component {
   renderMotorInputs(from, to) {
     return this.props.uiproperties.components
       .slice(from, to)
+      .filter(
+        ({ attribute }) =>
+          this.props.hardwareObjects[attribute].type === 'MOTOR',
+      )
```

Comment:
Just wondering if this filtering should be done earlier on (maybe in the parent component), so that `this.renderMotorInputs(0, 3)` is always guaranteed to render the first 3 motors?

Reply:
ah, yes you are right it could be an issue. Well spotted Ill try to fix that !

Reply:
I tired to do something nicer, by passing only the motor ui attributes and the motor hardware objects to the `MotorControls` component.

Reply:
Looks great :100: